transport/conn.go

```diff
@@ -166,10 +171,26 @@ type connHeader struct {
 // As a side effect, the peer's protocol number is stored in the conn.
 // Also, various properties are initialized.
 func (p *conn) handshake() error {
+
+	// Rational for a timeout mechanism (which may need to move elsewhere)
+	// The dial may have worked but that does not guarantee the
+	// server is going to send us any data.
+	// No/partial data will cause binary.Read to block indefinitely
+	// if the socket remains open. Closing the dialer does not work
+	// because the handshake is not complete (check for p.open).
+
+	// TODO: Timeout should be configurable however that results in a
```

Comment:
I don't think this should be configurable.  The handshake should go quickly, and 5 seconds is more than enough time to deal with even the crappiest networks.

Reply:
HOWEVER, I don't think we want to keep this deadline in place once the handshake is completed.  Because we could have large BULK transfers (think a 4 GB message, which is perfectly legal) that might take quite a while to send.  Or the connection might be backed up due to other large messages in the pipe ahead of it.  Therefore we need to ensure that the timeout does not persist once the handshake is complete.

Reply:
Right. That is why I defer SetDeadline to 0. That should disable the current deadline per docs "The deadline applies to all future and pending I/O". I figure the deadline set during handshake is done right after dialer connection creation which is prior to the conn being available for transport usage. After the dial/handshake completes other deadlines could be set on it specific to how mangos wants to use the connection.